packages/@aws-cdk/cdk/lib/stack.ts

```diff
@@ -159,10 +142,10 @@ export class Stack extends Construct implements ITaggable {
     this.configuredEnv = props.env || {};
     this.env = this.parseEnvironment(props.env);
 
-    this.logicalIds = new LogicalIDs(props && props.namingScheme ? props.namingScheme : new HashedAddressingScheme());
+    this.logicalIds = new LogicalIDs(props.namingScheme ? props.namingScheme : new HashedAddressingScheme());
     this.name = props.stackName !== undefined ? props.stackName : this.calculateStackName();
-    this.autoDeploy = props && props.autoDeploy === false ? false : true;
-    this.tags = new TagManager(TagType.KeyValue, "aws:cdk:stack", props.tags);
+    this.entrypoint = props.entrypoint === undefined ? true : false;
```

Comment:
I'm not a fan of "undefined is truthy" here. Also - it strikes me that `entrypoint` denotes the tip of a dependency graph that has to be fully deployed, but `autoDeploy` also meant that a given stack would be ignored from deploy unless explicitly requested, even if it is part of the dependency closure... I'm worried that we're going to be confusing here :/